### DISCUSSION TITLE
Setting env var matching tag name

### DISCUSSION THREAD

**Discussion by robertsLando** [docker/buildx#1322] [Q&A] [answered]
I'm actually pushing images using different tags to different registries. I would like to set an env var (or else) that matches actual image tag name so I can get it from inside container without the need to map docker socket. Is there a way to do this?

**Comment by jedevc** [comment, marked as answer]
You could create a build arg for this purpose, placing it at the bottom of your Dockerfile:

```
ARG image
```

Then during build, you can set this with `--build-arg` - because the `ARG` comes last, I think cache should be preserved between different runs of `build`, so the subsequent builds should be almost instant (except for the time to push to the new registry). 

Unfortunately, to get the data, you will need to build multiple times - since this requires changing the runtime environment config, it can't be done without it.

**Reply by robertsLando** [reply]
Thanks @jedevc !
